Source/WebCore/Modules/mediasource/MediaSource.cpp

```diff
@@ -509,8 +509,10 @@ ExceptionOr<void> MediaSource::setDurationInternal(const MediaTime& duration)
     // 4. If the new duration is less than old duration, then call remove(new duration, old duration)
     // on all objects in sourceBuffers.
     if (m_duration.isValid() && newDuration < m_duration) {
-        for (auto& sourceBuffer : *m_sourceBuffers)
-            sourceBuffer->rangeRemoval(newDuration, m_duration);
+        for (auto& sourceBuffer : *m_sourceBuffers) {
+            if (auto length = sourceBuffer->bufferedInternal().length())
+                sourceBuffer->rangeRemoval(newDuration, sourceBuffer->bufferedInternal().ranges().end(length - 1));
```

Comment:
When setDurationInternal is called duration could be small, but buffered much more.
But when endOfStream is called
MediaSource.cpp:604  highest end time will be used as duration, but these buffers should be removed due to duration already changed manually?

        // 1. Run the duration change algorithm with new duration set to the highest end time reported by
        // the buffered attribute across all SourceBuffer objects in sourceBuffers.